src/components/hover.tsx

```diff
@@ -3,6 +3,7 @@ import "./hover.css";
 
 interface HoverProps {
   children: ReactNode;
+  action: Function;
```

Comment:
Love the use of TS! It's typically best practice to be as specific as possible with type definitions, especially for functions. In this context, I think the type we'll want to use is
```suggestion
  action: (newValue: string | null) => void;
```
That tells your function that it needs to pass in either string or null and it shouldn't expect anything to be returned